### ISSUE TITLE
is it possible to build a windows version ?

### ISSUE THREAD

**Issue by AlbertG123** [tpm2-software/tpm2-pkcs11#782]
hi, 
is it possible to build a windows version?  so I want to build app -> app will invoke pkcs11 interface ->pkcs11 is based on local TPM2.0.
or I can build a linux version and deploy it as a Linux server tool.  and my windows app should connect to Linux server to use the pkcs11 interface ?

**Comment by williamcroberts** [comment]
> hi, is it possible to build a windows version? so I want to build app -> app will invoke pkcs11 interface ->pkcs11 is based on local TPM2.0.

Yes it should be feasible, you may need to do some work on porting the lockfile logic db.c to windows filesystem C api's.

> or I can build a linux version and deploy it as a Linux server tool. and my windows app should connect to Linux server to use the pkcs11 interface ?

I'm not really sure what that a Linux server tool is, but that seems like overkill compared to just porting the lockfile logic.

**Comment by gowdamanil** [comment]
hello @AlbertG123 

Did you port the above changes suggested by williamcroberts for windows?

If yes, can you guide me through it?

**Comment by AlbertG123** [comment]
Not yet, I don't work on it now.

From: gowdamanil ***@***.***>
Sent: Saturday, November 2, 2024 8:42 PM
To: tpm2-software/tpm2-pkcs11 ***@***.***>
Cc: Albert Gao ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [tpm2-software/tpm2-pkcs11] is it possible to build a windows version ? (Issue #782)


hello @AlbertG123<https://github.com/AlbertG123>

Did you port the above changes suggested by williamcroberts for windows?

If yes, can you guide me through it?

-
